test/common.py

```diff
@@ -39,6 +39,7 @@
 parser = argparse.ArgumentParser(add_help=False)
 parser.add_argument('--seed', type=int, default=1234)
 parser.add_argument('--accept', action='store_true')
+parser.add_argument('--shard', type=int, required=False)
```

Comment:
This is a bit annoying because it will be a valid argument to all tests, but only have any meaning in test_nn

Reply:
@yf225 you can do this in `test_nn.py`

```py
from common import parser
parser.add_argument(...)
```

You can also wrap the argparser in a class and subclass it in `test_nn.py` if that style feels nicer.